src/modules/dropbox.js

```diff
@@ -1,213 +1,216 @@
+			root: 'sandbox',
 
-		// Map DELETE requests 
-		del : {
-			"me/files" : "fileops/delete?root=@{root|sandbox}&path=@{id}",
-			"me/folder" : "fileops/delete?root=@{root|sandbox}&path=@{id}"
-		},
+			// Map GET requests
+			get: {
+				me: 'account/info',
 
+				// https://www.dropbox.com/developers/core/docs#metadata
+				'me/files': req('metadata/@{root|sandbox}/@{parent}'),
+				'me/folder': req('metadata/@{root|sandbox}/@{id}'),
+				'me/folders': req('metadata/@{root|sandbox}/'),
 
-		wrap : {
-			me : function(o){
-				formatError(o);
-				if(!o.uid){
-					return o;
+				default: function(p, callback) {
+					if (p.path.match('https://api-content.dropbox.com/1/files/')) {
+						// this is a file, return binary data
+						p.method = 'blob';
+					}
+
+					callback(p.path);
 				}
-				o.name = o.display_name;
-				o.first_name = o.name.split(" ")[0];
-				o.last_name = o.name.split(" ")[1];
-				o.id = o.uid;
-				delete o.uid;
-				delete o.display_name;
-				return o;
 			},
-			"default"	: function(o,headers,req){
-				formatError(o);
-				if(o.is_dir && o.contents){
-					o.data = o.contents;
-					delete o.contents;
-
-					for(var i=0;i<o.data.length;i++){
-						o.data[i].root = o.root;
-						format_file(o.data[i],headers,req);
+
+			post: {
+				'me/files': function(p, callback) {
+
+					var path = p.data.parent;
+					var fileName = p.data.name;
+
+					p.data = {
+						file: p.data.file
+					};
+
+					// Does this have a data-uri to upload as a file?
+					if (typeof (p.data.file) === 'string') {
+						p.data.file = hello.utils.toBlob(p.data.file);
 					}
-				}
 
-				format_file(o,headers,req);
+					callback('https://api-content.dropbox.com/1/files_put/@{root|sandbox}/' + path + '/' + fileName);
+				},
+
+				'me/folders': function(p, callback) {
+
+					var name = p.data.name;
+					p.data = {};
 
-				if(o.is_deleted){
-					o.success = true;
+					callback('fileops/create_folder?root=@{root|sandbox}&' + hello.utils.param({
+						path: name
+					}));
 				}
+			},
 
-				return o;
-			}
-		},
-
-		// doesn't return the CORS headers
-		xhr : function(p){
-
-			// the proxy supports allow-cross-origin-resource
-			// alas that's the only thing we're using. 
-			if( p.data && p.data.file ){
-				var file = p.data.file;
-				if( file ){
-					if(file.files){
-						p.data = file.files[0];
+			// Map DELETE requests
+			del: {
+				'me/files': 'fileops/delete?root=@{root|sandbox}&path=@{id}',
+				'me/folder': 'fileops/delete?root=@{root|sandbox}&path=@{id}'
+			},
+
+			wrap: {
+				me: function(o) {
+					formatError(o);
+					if (!o.uid) {
+						return o;
+					}
+
+					o.name = o.display_name;
+					o.first_name = o.name.split(' ')[0];
+					o.last_name = o.name.split(' ')[1];
+					o.id = o.uid;
+					delete o.uid;
+					delete o.display_name;
+					return o;
+				},
+
+				default: function(o, headers, req) {
```

Comment:
Yup, IE8 throws a syntax error.

Its easy to replace "default" with "*" etc, but requires a core change, so best left to another commit.